source/backend/api/Areas/Documents/DocumentQueueController.cs

```diff
@@ -33,32 +42,155 @@ public class DocumentQueueController : ControllerBase
+        /// <returns></returns>
+        [HttpPost("{documentQueueId:long}/upload")]
+        [HasPermission(Permissions.SystemAdmin)]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(List<DocumentTypeModel>), 200)]
+        [SwaggerOperation(Tags = new[] { "document-types" })]
+        [TypeFilter(typeof(NullJsonResultFilter))]
+        public async Task<IActionResult> Upload(long documentQueueId, [FromBody] DocumentQueueModel documentQueue)
+        {
+            _logger.LogInformation(
+                "Request received by Controller: {Controller}, Action: {ControllerAction}, User: {User}, DateTime: {DateTime}",
+                nameof(DocumentQueueController),
+                nameof(Upload),
+                User.GetUsername(),
+                DateTime.Now);
+
+            documentQueue.ThrowIfNull(nameof(documentQueue));
+            if (documentQueueId != documentQueue.Id)
+            {
+                throw new BadRequestException("Invalid document queue id.");
+            }
+
+            var queuedDocuments = await _documentQueueService.Upload(_mapper.Map<PimsDocumentQueue>(documentQueue));
+            var updatedDocumentQueue = _mapper.Map<DocumentQueueModel>(queuedDocuments);
+            return new JsonResult(updatedDocumentQueue);
+        }
+
         /// <summary>
         /// Search for Document Queue items via filter.
         /// </summary>
         /// <returns></returns>
-        [HttpGet("search")]
+        [HttpPost("search")]
         [HasPermission(Permissions.SystemAdmin)]
         [Produces("application/json")]
-        [ProducesResponseType(typeof(List<Models.Concepts.Document.DocumentTypeModel>), 200)]
+        [ProducesResponseType(typeof(List<DocumentTypeModel>), 200)]
         [SwaggerOperation(Tags = new[] { "document-types" })]
         [TypeFilter(typeof(NullJsonResultFilter))]
-        public IActionResult GetDocumentTypes([FromBody] DocumentQueueFilter filter)
+        public IActionResult SearchQueuedDocuments([FromBody] DocumentQueueFilter filter)
         {
+            _logger.LogInformation(
+                "Request received by Controller: {Controller}, Action: {ControllerAction}, User: {User}, DateTime: {DateTime}",
+                nameof(DocumentQueueController),
+                nameof(SearchQueuedDocuments),
+                User.GetUsername(),
+                DateTime.Now);
+
             var queuedDocuments = _documentQueueService.SearchDocumentQueue(filter);
-            var documentQueueModels = _mapper.Map<List<Models.Concepts.Document.DocumentQueueModel>>(queuedDocuments);
+            var documentQueueModels = _mapper.Map<List<DocumentQueueModel>>(queuedDocuments);
             return new JsonResult(documentQueueModels);
         }
 
+        /// <summary>
+        /// Get Document Queue item via id.
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet("{documentQueueId:long}")]
+        [HasPermission(Permissions.SystemAdmin)]
```

Comment:
general question about permissions - I see all queue endpoints require sysadmin permissions. Is this intended? How will the frontend connect to this new API for non-admin users?

Reply:
the frontend won't connect to this api, at least, not directly. The documents will come in from the frontend via the document controller, which will handle adding the document queue entity. This controller that allows direct interaction with the document queue is really only used by the scheduler.